catalog/spatial/spatial-commands/src/main/java/org/codice/ddf/commands/spatial/gazetteer/GazetteerBuildSuggesterIndexCommand.java

```diff
@@ -0,0 +1,109 @@
+import static ddf.catalog.Constants.SUGGESTION_DICT_KEY;
+import static ddf.catalog.Constants.SUGGESTION_QUERY_KEY;
+import static ddf.catalog.data.types.Core.METACARD_TAGS;
+import static org.codice.ddf.spatial.geocoding.GeoCodingConstants.GAZETTEER_METACARD_TAG;
+import static org.codice.ddf.spatial.geocoding.GeoCodingConstants.SUGGEST_PLACE_KEY;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.codice.ddf.commands.catalog.SubjectCommands;
+
+@Service
+@Command(
+  scope = "gazetteer",
+  name = "build-suggester-index",
+  description =
+      "Builds the Solr suggest index used for placename autocompletion in Intrigue when using "
+          + "the offline gazetteer. This index is built automatically when the gazetteer is "
+          + "updated using the Admin UI or the `gazetteer:update` command, but if that build "
+          + "attempt fails then this command can be used to attempt to build it again."
+)
+public class GazetteerBuildSuggesterIndexCommand extends SubjectCommands {
+  @Reference private CatalogFramework catalogFramework;
+
+  @Reference private FilterBuilder filterBuilder;
+
+  public void setCatalogFramework(CatalogFramework catalogFramework) {
+    this.catalogFramework = catalogFramework;
+  }
+
+  public void setFilterBuilder(FilterBuilder filterBuilder) {
+    this.filterBuilder = filterBuilder;
+  }
+
+  @Override
+  protected Object executeWithSubject() {
+    console.println("Building the gazetteer suggester index.");
+    console.println(
+        "This will take a few seconds or longer depending on how many gazetteer entries there are.");
+
+    buildSuggesterIndex();
+
+    console.println("Done.");
+
+    return null;
+  }
+
+  private void buildSuggesterIndex() {
+    final QueryRequest buildSuggesterIndex = getQueryBuildSuggesterIndex();
+
+    requestBuildSuggesterIndex(buildSuggesterIndex);
+  }
+
+  private void requestBuildSuggesterIndex(QueryRequest buildSuggesterIndex) {
+    try {
+      catalogFramework.query(buildSuggesterIndex);
+    } catch (SourceUnavailableException | FederationException | UnsupportedQueryException e) {
+      printErrorMessage(
+          "Could not build the Solr suggestion index. Try running this command again after resolving the following error.");
+      printErrorMessage("Error: " + e.getMessage());
```

Comment:
Maybe it's not typical for the interactive commands, but would it be worth logging or printing the stacktrace?

Reply:
Logging seems appropriate since it could indicate a problem with the system.